src/main/kotlin/uk/gov/justice/digital/hmpps/authorizationserver/config/AuthorizationServerConfig.kt

```diff
@@ -63,6 +64,10 @@ class AuthorizationServerConfig(
   @Value("\${jwt.jwk.key.id}") private val keyId: String,
   @Value("\${server.base-url}") private val baseUrl: String,
   @Value("\${server.servlet.context-path}") private val contextPath: String,
+  @Value("#{'\${jwt.auxiliary.jwk.key.id}' <= '' ? null : '\${jwt.auxiliary.jwk.key.id}'}") private val keyIdAuxiliary: String?,
```

Comment:
I'm wondering if you could avoid needing to specify this configuration in two places...

Could the KeyPairAccessor become the JWKKeyAccessor and instead of returning key pairs, it returns keys, so it would offer:

getPrimaryPublicKey(): RSAKey
getPrimaryPrivateKey(): RSAKey
getAuxiliaryPublicKey(): RSAKey?

(no need for the aux private key)

So no jwt config would need to exist in this file, the logic to pull particular types of key would be pushed down into the accessor.

Reply:
You could add some helper functions to the JWKKeyAccessor to prevent duplication of code between the primary and auxiliary versions too

Reply:
Thanks Jon..Updated